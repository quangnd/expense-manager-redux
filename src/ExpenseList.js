import React from "react";
import { connect } from "react-redux";

const ExpenseList = props => (
  <div>
    <h1>ExpenseList {props.expenses.length}</h1>
  </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: state.expenses,
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseList)
