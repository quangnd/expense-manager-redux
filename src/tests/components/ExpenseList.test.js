import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../ExpenseList";
import expenses from '../fixtures/expenses';

test('should render ExpenseList with expenses', () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />)
  expect(wrapper).toMatchSnapshot()
})

test('should render ExpenList with empty data', () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />)
  expect(wrapper).toMatchSnapshot()
})
