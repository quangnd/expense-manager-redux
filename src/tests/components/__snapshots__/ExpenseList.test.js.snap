// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenList with empty data 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="water bill"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={11000}
    createdAt={-345600000}
    description="electric bill"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4400}
    createdAt={345600000}
    description="internet monthly"
    id="3"
    key="3"
    note=""
  />
</div>
`;
